Sally7/Internal/NetworkStreamExtensions.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Net.Sockets;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Sally7.Internal
+{
+    internal static class NetworkStreamExtensions
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
I took inspiration from [RabbitMQ .NET Client](https://github.com/rabbitmq/rabbitmq-dotnet-client), which uses this in quite a lot. I'm not sure what to do with your comment though (I actually expected you might comment on it), would your suggestion be to remove it an let the compiler figure it out, or would you say it doesn't hurt either?

Reply:
I'd remove that attribute.
* the JIT will inline this small method pretty sure
* makes the code cleaner

In general I apply _AggressiveInlining_ only when benchmarks + looking at codegen show it's beneficial. Otherwise leave the JIT alone with it's decisions.

Reply:
I generally only benchmark or look at codegen when stuff breaks 😆 But to that regard, I was actually recently looking into [ZeroMock](https://github.com/ZeroMock/ZeroMock) so I could mock the `TcpClient` / `Socket` in order to add unit tests without wrapping the socket implementation, [unfortunately that didn't work out](https://github.com/ZeroMock/ZeroMock/issues/15).